src/widgets/widget-model.js

```diff
@@ -26,7 +26,7 @@ module.exports = cdb.core.Model.extend({
   update: function (changes) {
     var attrs = _.pick(changes, this.get('attrsNames'));
     this.set(attrs);
-    attrs = _.pick(changes, this.get('dataviewModelAttrsNames'));
+    attrs = _.pick(changes, this.dataviewModel.constructor.ATTRS_NAMES);
```

Comment:
I just realized… we could actually do the same kind of solution on the dataview models, i.e. have a `dataviewModel.update(attrs)` method, that only picks the attrs that it needs and discard the rest. This way  `dataviewModel.constructor.ATTRS_NAMES` can remain private to cartodb.js